app/controllers/graders_controller.rb

```diff
@@ -122,11 +122,18 @@ def global_actions
             found_empty_submission = true
           end
         end
-        if found_empty_submission
-          randomly_assign_graders(filtered_grouping_ids, grader_ids)
-          flash_now(:info, I18n.t('graders.group_submission_no_files'))
-        else
-          randomly_assign_graders(grouping_ids, grader_ids)
+        begin
+          weights = params[:weightings].map { |weight| Float(weight) }
+          if weights.reduce(:+) == 0 || weights.select(&:negative?).length > 0
+            head :bad_request
+          elsif found_empty_submission
+            randomly_assign_graders(filtered_grouping_ids, grader_ids, weights)
+            flash_now(:info, I18n.t('graders.group_submission_no_files'))
+          else
+            randomly_assign_graders(grouping_ids, grader_ids, weights)
+          end
+        rescue StandardError
+          head :bad_request
```

Comment:
Same comments as above. Note that you can grab the exception object and then just flash the `message` of that exception.